policies/registry.rego

```diff
@@ -89,10 +89,82 @@ deny contains attr_registry_violation(description, group.id, "") if {
     description := sprintf("Semconv group '%s' does not contain stability field. All semconv definitions must include stability level.", [group.id])
 }
 
+
+# check that attribute is not defined or referenced more than once within the same group
+deny contains attr_registry_violation(description, group.id, name) if {
+    group := input.groups[_]
+    attr := group.attributes[_]
+    name := attr.name
+
+    collisions := [n | n := group.attributes[_].name; n == name ]
+    count(collisions) > 1
+
+    description := sprintf("Attribute '%s' is already defined in the group '%s'. Attributes must be unique.", [name, group.id])
+}
+
+# check that member ids do not collide within the same attribute
+deny contains attr_registry_violation(description, group.id, attr.id) if {
+    group := input.groups[_]
+    startswith(group.id, "registry.")
+
+    attr := group.attributes[_]
+    member := attr.type.members[_]
+
+    collisions := [n | n := attr.type.members[_].id; n == member.id ]
+    count(collisions) > 1
+
+    description := sprintf("Member with id '%s' is already defined on the attribute '%s' in the group '%s'. Member id must be unique.", [member.id, attr.id, group.id])
+}
+
+# check that member values do not collide within the same attribute
+deny contains attr_registry_violation(description, group.id, attr.id) if {
+    group := input.groups[_]
+    startswith(group.id, "registry.")
+    attr := group.attributes[_]
+    member := attr.type.members[_]
+    not is_property_set(member, "deprecated")
```

Comment:
I can never understand the difference between `not obj.prop` vs `obj.prop == null` vs this hack. For whatever reason this works and `not obj.prop`,  `obj.prop == null` don't 😿